src/looprestoration_tmpl_8.rs

```diff
@@ -1037,7 +1061,70 @@ unsafe extern "C" fn loop_restoration_dsp_init_x86(
         (*c).sgr[2] = Some(dav1d_sgr_filter_mix_8bpc_avx512icl);
     }
 }
-
+#[cfg(all(feature = "asm", target_arch = "arm"))]
+unsafe extern "C" fn wiener_filter_neon(
+    dst: *mut pixel,
+    stride: ptrdiff_t,
+    left: *const [pixel; 4],
+    mut lpf: *const pixel,
+    w: libc::c_int,
+    h: libc::c_int,
+    params: *const LooprestorationParams,
+    edges: LrEdgeFlags,
+) {
+    let filter: *const [int16_t; 8] = (*params).filter.0.as_ptr();
+    let mut mid: Align16<[int16_t; 68 * 384]> = Align16([0; 68 * 384]);
+    let mut mid_stride: libc::c_int = w + 7 & !7;
+    // Horizontal filter
+    dav1d_wiener_filter_h_8bpc_neon(
+        &mut *mid.0.as_mut_ptr().offset((2 * mid_stride) as isize),
+        left,
+        dst,
+        stride,
+        filter.offset(0),
```

Comment:
This would be translated as `(*filter.offset(0)).as_ptr()` when the `fh` arg is `*const int16_t`.